ClientLib/src/main/java/com/o3dr/android/client/utils/data/tlog/TLogParser.java

```diff
@@ -0,0 +1,349 @@
+package com.o3dr.android.client.utils.data.tlog;
+
+import android.os.Handler;
+import android.util.Log;
+
+import com.MAVLink.MAVLinkPacket;
+import com.MAVLink.Messages.MAVLinkMessage;
+import com.MAVLink.Parser;
+
+import java.io.BufferedInputStream;
+import java.io.DataInputStream;
+import java.io.EOFException;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.net.URI;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * Parse TLog file into Events
+ */
+public class TLogParser {
+    private static final String LOG_TAG = TLogParser.class.getSimpleName();
+
+    private static final Parser parser = new Parser();
+    private static ExecutorService parseTlogExecutor;
+
+    public static class TLogIterator {
+        private File file;
+        private DataInputStream in = null;
+        private ExecutorService iteratorExecutor;
+        private final Handler handler;
+
+        /**
+         * Constructor to create a TLogIterator with new Handler.
+         *
+         * @param uri Location of the TLog files
+         */
+        public TLogIterator(URI uri) {
+            this(uri, new Handler());
+        }
+
+        /**
+         * Constructor to create a TLogIterator with a specified Handler.
+         *
+         * @param uri     Location of the TLog files
+         * @param handler Handler to post results to
+         */
+        public TLogIterator(URI uri, Handler handler) {
+            this.handler = handler;
+            file = new File(uri.toString());
+        }
+
+        /**
+         * Opens TLog file to begin iterating.
+         *
+         * @throws FileNotFoundException
+         */
+        public void start() throws FileNotFoundException {
+            iteratorExecutor = Executors.newSingleThreadExecutor();
+            in = new DataInputStream(new BufferedInputStream(new FileInputStream(file)));
+        }
+
+        /**
+         * Closes TLog file
+         *
+         * @throws IOException
+         */
+        public void finish() throws IOException {
+            iteratorExecutor.shutdownNow();
+            in.close();
+        }
+
+        /**
+         * Retrieve next message from TLog file asynchronously.
+         * start must be called before this method.
+         *
+         * @param callback
+         */
+        public void nextAsync(final TLogIteratorCallback callback) {
```

Comment:
@chaviw have this method delegated to the `nextAsync(TLogIteratorFilter filter, TLogIteratorCallback callback)` method by passing a filter that accepts all events.